src/track/track.cpp

```diff
@@ -128,6 +128,15 @@ void Track::importMetadata(
     const auto newBpm = importedMetadata.getTrackInfo().getBpm();
     const auto newKey = importedMetadata.getTrackInfo().getKey();
     const auto newReplayGain = importedMetadata.getTrackInfo().getReplayGain();
+#ifdef __EXTRA_METADATA__
+    const auto newSeratoTags = importedMetadata.getTrackInfo().getSeratoTags();
+    {
+        // enter locking scope
+        QMutexLocker lock(&m_qMutex);
```

Comment:
This must be done within the common lock scope, not independently.

Moreover the Serato tags should depend on the `Track` object as a cyclic dependency. The `Track` object may depend on the Serator tags, but not vice versa.

Reply:
The Serato tags as a component of `TrackInfo` are not allowed to know anything including and above `TrackInfo`. If you need some properties from `TrackInfo` you need to pass them separately as parameters downstream.